### PR TITLE
Bug 550659 - Extendable Footer actions for the Dialog(Opal)

### PR DESCRIPTION
**Pull request by sibeon**
I added a toolbar to the footer to be able to create multiple tool items. This makes it very flexible to add additional actions to the dialog.

Signed-off-by: Stefan Nöbauer <stefan.noebauer@kgu-consulting.com>

### CODE REVIEW

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java [hunk 1]
```diff
@@ -99,18 +109,27 @@ void render() {
 		composite.setBackground(getGreyColor());
 
 		int numberOfColumns = buttonLabels == null ? 0 : buttonLabels.size();
-		if (details) {
-			numberOfColumns += 2;
+		if (details | !footerActions.isEmpty()) {
```
Comment (lcaron) [inline widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: Please use double pipe : if (details || !footerActions.isEmpty())

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java [hunk 2]
```diff
@@ -99,18 +109,27 @@ void render() {
...
+			numberOfColumns += 1;
 		}
 
 		final GridLayout gridLayout = new GridLayout(numberOfColumns, false);
 		gridLayout.marginHeight = gridLayout.marginWidth = 10;
 		composite.setLayout(gridLayout);
 
-		if (details) {
+		if (details | !footerActions.isEmpty()) {
```
Comment (lcaron) [inline widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: Please use double pipe

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java [hunk 3]
```diff
@@ -203,50 +222,66 @@ public void run() {
...
-		final Label icon = new Label(composite, SWT.NONE);
-		icon.setBackground(getGreyColor());
-		icon.setImage(isExpanded() ? getFewerDetailsImage() : getMoreDetailsImage());
-		icon.setLayoutData(new GridData(GridData.CENTER, GridData.CENTER, false, false));
-
-		final Label label = new Label(composite, SWT.NONE);
-		label.setBackground(getGreyColor());
-		label.setText(isExpanded() ? expandedLabelText : collapsedLabelText);
-		label.setLayoutData(new GridData(GridData.BEGINNING, GridData.CENTER, false, false));
+		ToolItem detailsItem = new ToolItem(toolbar, SWT.NONE);
```
Comment (lcaron) [inline widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: Please add final keyword before declaring the variable

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java [hunk 4]
```diff
@@ -203,50 +222,66 @@ public void run() {
...
 		};
 
-		label.addListener(SWT.MouseUp, listener);
-		icon.addListener(SWT.MouseUp, listener);
+		detailsItem.addListener(SWT.Selection, listener);
+	}
+
+	private void createFooterActions() {
+		for (FooterAction action : footerActions) {
+			ToolItem item = new ToolItem(toolbar, SWT.NONE);
```
Comment (lcaron) [inline widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: Please add final keyword

**Review by lcaron — COMMENTED** [review]
A few problems to fix, good job !

**Review by lcaron — CHANGES_REQUESTED** [review]
Please make the modifications so we can merge it

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java (hunk 4 above)
Comment (sibeon) [reply widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: done

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java (hunk 3 above)
Comment (sibeon) [reply widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: done

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java (hunk 2 above)
Comment (sibeon) [reply widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: done

File: widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java (hunk 1 above)
Comment (sibeon) [reply widgets/opal/dialog/org.eclipse.nebula.widgets.opal.dialog/src/org/eclipse/nebula/widgets/opal/dialog/FooterArea.java]: done

**Review by lcaron — APPROVED** [review]

**Pull request merged**